CMP1903M A01 2223/Pack.cs

```diff
@@ -8,121 +8,123 @@ namespace CMP1903M_A01_2223
 {
     class Pack
     {
+        //fields
         public static List<Card> pack = new List<Card>();
         public static List<Card> dealtCards = new List<Card>();
-        public static List<Card> packFirstHalf = new List<Card>();
-        public static List<Card> packSecondHalf = new List<Card>();
+        public static List<Card> packHalf = new List<Card>();
+      
 
         // Pack Constructor
         public Pack()
         {
-            int Card_suit = 0;
-            int Card_value = 0;
-            int Num_Of_Cards = 1;
-            while (Num_Of_Cards <= 52)
+
+            int Card_suit = 0; // this keeps track of what suit it is
+            int Card_value = 0; // this keeps track of the cards value
+            int Num_Of_Cards = 1; //This keeps track of how many cards have been created
+            while (Num_Of_Cards <= 52) // While there is less than the number of cards in a pack this will run
             {
-                pack.Add(new Card(Card_value, Card_suit));
-                Card_value++;
-                Num_Of_Cards++;
-                if (Card_value == 13)
+                pack.Add(new Card(Card_value, Card_suit)); // This adds the card to the pack depending on the current value and suit
+                Card_value++; // This adds to the card value
+                Num_Of_Cards++; // This adds to the number of cards made
+                if (Card_value == 13) // This makes it so that when it reaches the required amount of cards in the suit
                 {
-                    Card_value = 0;
-                    Card_suit++;
+                    Card_value = 0; // It will reset the card value to 0
+                    Card_suit++; // And add one to the suit so changes suit
                 }
             }
         }
+
         // Methods
         // This method shuffles the pack depending on which type of shuffle is selected
         public static bool shuffleCardPack(int typeOfShuffle)
         {
             // Fisher Yates shuffle
-            if (typeOfShuffle == 1)
+            if (typeOfShuffle == 1) // This checks if the type of shuffles value is one, if it is it will continue with the function
             {
-                var random = new Random();
-                for (int Num_Of_Cards = pack.Count; Num_Of_Cards > 1; Num_Of_Cards--)
+                var random = new Random(); // This sets the value of the random variable to a random number
```

Comment:
better to specify what type the variable is yourself rather than letting c# decide